src/rendering/renderers/shared/texture/RenderableGCSystem.ts

```diff
@@ -186,5 +191,17 @@ export class RenderableGCSystem implements System<RenderableGCSystemOptions>
         this.enabled = false;
         this._renderer = null as any as Renderer;
         this._managedRenderables.length = 0;
+        this._managedRenderablesIndexes = {};
+    }
+
+    private _removeRenderable(renderable: Container): void
+    {
+        const index = this._managedRenderablesIndexes[renderable.uid];
+
+        if (index !== null)
+        {
+            this._managedRenderables.splice(index, 1);
```

Comment:
If you splice here, wouldn't that invalidate the indexes of all the managed renderables ahead of it?

Reply:
you are correct, thats what i get for late night coding 😄